pandas/tests/frame/methods/test_align.py

```diff
@@ -129,6 +129,26 @@ def test_align_mixed_int(self, mixed_int_frame):
         )
         tm.assert_index_equal(bf.index, Index([]))
 
+    def test_align_categorical(self):
```

Comment:
can you rename this to test_align_categorical_unordered and then add an ordered version (alt is to parameterize this)

Reply:
I guess the same holds as previous, if one categorical is ordered and the other isn't, the expected `dtype` is object?